routes/authors.js

```diff
@@ -1,8 +1,103 @@
 const express = require("express");
-const generateId = require("uuid/v4");
-
 const router = express.Router();
+const generateId = require("uuid/v4");
+const authors = require("../data/authors");
+const books = require("../data/books");
 
 // Define your endpoints here
 
+// GET ALL AUTHORS & SORT BY NAME
+router.get('/', (req, res, next) => {
+    console.log(req.query)
+    if(req.query.sort === 'name'){
+        const sortedArray= authors.slice(0);
+        
+            sortedArray.sort((a,b) => {
+                console.log('hello');
+                return a.name < b.name ? -1 : 1 ;
+            });
+            res.json(sortedArray);
+    } else {
+        res.json({authors});
+    }
+});
+
+// ADD AN AUTHOR
+router.post('/', (req, res, next) => {
+    const {name, website} = req.body;
+    const id = generateId();
+    const newAuthor = {name, website, id};
+    authors.push(newAuthor);
+    res.json(newAuthor);
+});
+
+// GET AUTHOR BY ID
+router.get('/:id', (req, res, next) => {
+    const id = req.params.id;
+    const foundAuthor = authors.find(obj => obj.id === id);
+    res.json(foundAuthor);
```

Comment:
What happens if you don't find an author? The proper thing to do would be to send a response with a code 404.